cpp/models/abm/parameters.h

```diff
@@ -25,16 +25,48 @@
 #include "abm/virus_variant.h"
 #include "abm/vaccine.h"
 #include "abm/test_type.h"
+#include "memilio/config.h"
+#include "memilio/io/auto_serialize.h"
+#include "memilio/io/io.h"
+#include "memilio/math/time_series_functor.h"
 #include "memilio/utils/custom_index_array.h"
 #include "memilio/utils/uncertain_value.h"
 #include "memilio/utils/parameter_set.h"
 #include "memilio/epidemiology/age_group.h"
 #include "memilio/epidemiology/damping.h"
 #include "memilio/epidemiology/contact_matrix.h"
+
+#include <algorithm>
 #include <limits>
+#include <type_traits>
 
 namespace mio
 {
+
+template <class IOContext, class T,
+          std::enable_if_t<std::is_same_v<UniformDistribution<double>::ParamType, T>, void*> = nullptr>
+void serialize_internal(IOContext& io, const T& p)
```

Comment:
I left it here since it only works with double, not with any FP, and it is only used for the ABM params. But maybe that is ok?

Reply:
So I found a way to make it work with generic Real type, but it is somewhat worse as it needs 4 template arguments (the fourth is mostly needed to differentiate from other enable_ifs, but it may become necessary to differentiate from other distribution adapters)
```cpp
template <class IOContext, class UniformDistributionParams,
          class ResultType = typename UniformDistributionParams::DistType::ResultType,
          std::enable_if_t<std::is_same_v<UniformDistributionParams, typename UniformDistribution<ResultType>::ParamType>,
                           void*> = nullptr>
```